File: 백준/Silver/10845. 큐/큐.py
from collections import deque

T = int(input())
arr = deque([])
for _ in range(T):
    direction = input()

    if direction[0] == 's':
        print(len(arr))
    elif direction[0] == 'e':
        print(0 if arr else 1)
    elif direction[0] == 'f':
        print(arr[0] if arr else -1)
    elif direction[0] == 'b':
        print(arr[-1] if arr else -1)
    elif direction == 'pop':
        print(arr.popleft() if arr else -1)
    else:
        _, x = direction.split(' ')
        arr.append(int(x))

File: 백준/Silver/10845. 큐/큐.java
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        sc.nextLine();

        Deque<Integer> deque = new ArrayDeque<>();

        for (int i=0; i<T; i++) {
            String s = sc.nextLine();

            if (s.startsWith("push")) {
                String[] parts = s.split(" ");
                int x = Integer.parseInt(parts[1]);
                deque.add(x);
            } else if (s.equals("pop")) {
                System.out.println(deque.isEmpty() ? -1 : deque.removeFirst());
            } else if (s.equals("size")) {
                System.out.println(deque.size());
            } else if (s.equals("empty")) {
                System.out.println(deque.isEmpty() ? 1 : 0);
            } else if (s.equals("front")) {
                System.out.println(deque.isEmpty() ? -1 : deque.getFirst());
            } else if (s.equals("back")) {
                System.out.println(deque.isEmpty() ? -1 : deque.getLast());
            }
        }

        sc.close();
    }
}